k2/csrc/fsa.h

```diff
@@ -74,7 +73,9 @@ struct Fsa {
   // Note: an index into the `arcs` array is called an arc-index.
   std::vector<Arc> arcs;
 
-  inline std::size_t NumStates() { return leaving_arcs.size(); }
+  inline StateId NumStates() {
```

Comment:
I'm wondering whether `inline` is needed here since the method is
defined inside the `struct` which is impicitly inline.

The method should be marked as `const`.

```cpp
StateId NumStates() const
```

Reply:
inline in C++ means "this symbol can appear multiple times, in different .o files". It has nothing to do with function inlining. You are correct that it can be removed in this case, because NumStates() is defined fully within the struct's declaration.

Reply:
sure